pkg/installation/util.go

```diff
@@ -62,22 +35,23 @@ func pluginVersionFromPath(installPath, pluginPath string) (string, error) {
 	return elems[1], nil
 }
 
-func getPluginVersion(p index.Platform) (version, uri string) {
-	return strings.ToLower(p.Sha256), p.URI
-}
-
-func getDownloadTarget(index index.Plugin) (version, uri string, fos []index.FileOperation, bin string, err error) {
+func getDownloadTarget(index index.Plugin) (version, sha256sum, uri string, fos []index.FileOperation, bin string, err error) {
+	// TODO(ahmetb): We have many return values from this method, indicating
+	// code smell. More specifically we return all-or-nothing, so ideally this
+	// should be converted into a struct, like InstallOperation{} contains all
+	// the data needed to install a plugin.
 	p, ok, err := index.Spec.GetMatchingPlatform()
 	if err != nil {
-		return "", "", nil, p.Bin, errors.Wrap(err, "failed to get matching platforms")
+		return "", "", "", nil, p.Bin, errors.Wrap(err, "failed to get matching platforms")
 	}
 	if !ok {
-		return "", "", nil, p.Bin, errors.New("no matching platform found")
+		return "", "", "", nil, p.Bin, errors.New("no matching platform found")
 	}
-	version, uri = getPluginVersion(p)
-	glog.V(4).Infof("Matching plugin version is %s", version)
-
-	return version, uri, p.Files, p.Bin, nil
+	version = index.Spec.Version
+	uri = p.URI
+	sha256sum = p.Sha256
```

Comment:
The original version had a `ToLower(p.Sha256)`. Are you sure it's not needed?

Reply:
The fact that we're not sure points to a lack of tests issue.

I recall arguing that we should make casing of sha256sum mandatory and part of the validation. I think it's time to do that now.

Reply:
I've created #281 to address this.